src/scanner.c

```diff
@@ -3434,7 +3434,15 @@ yaml_parser_scan_plain_scalar(yaml_parser_t *parser, yaml_token_t *token)
 
             if (parser->flow_level
                     && CHECK(parser->buffer, ':')
-                    && !IS_BLANKZ_AT(parser->buffer, 1)) {
+                    && (
+                        CHECK_AT(parser->buffer, ',', 1)
+                        || CHECK_AT(parser->buffer, '?', 1)
+                        || CHECK_AT(parser->buffer, '[', 1)
+                        || CHECK_AT(parser->buffer, ']', 1)
+                        || CHECK_AT(parser->buffer, '{', 1)
+                        || CHECK_AT(parser->buffer, '}', 1)
+                    )
+                    ) {
```

Comment:
> Moreover, ? certainly is not a flow indicator, so I am unsure how it found its way in here.

I added this for consistency, because it's also in the next if-statement. Maybe I should take it out in both statements. Then `[foo:?bar]` would be allowed.

Reply:
> This change is undesirable. Regardless of how strange it is, `[foo:]]` is valid YAML 1.1

The point is that this currently is not allowed in libyaml, and I just keep it like that with this change. The only thing which changes is to allow `[foo:bar]`.
`ns-plain-char - ,[]{}` characters after the colon. So I don't know why this change would be undesirable.

Could you clarify what you mean?

The goal of this PR is to accept things that make sense *and* are valid, but leaving the other cases alone for now, as I wrote above.